### PR TITLE
unmountVolumes check shouldPodRuntimeBeRemoved

### PR DESCRIPTION
**Pull request by yingchunliu-zte**
Signed-off-by: yingchunliu-zte <liu.yingchun@zte.com.cn>

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
When kubelet restarts, it may fail to access apiserver and obtain PVC. As a result, DSW does not have a volume, and the volume is reconstructed, resulting in unnecessary umount operations. This is dangerous to applications.

@jingxu97 @gnufied @Jiawei0227 @dchen1107 @yangjunmyfm192085 @mrunalp @249043822

### CODE REVIEW

**Review by yangjunmyfm192085 — COMMENTED** [review]
/ok-to-test

**Comment by yingchunliu-zte** [comment]
> /retest

/retest

File: pkg/kubelet/volumemanager/reconciler/reconciler.go [hunk 1]
```diff
@@ -181,6 +184,10 @@ func (rc *reconciler) unmountVolumes() {
 	// Ensure volumes that should be unmounted are unmounted.
 	for _, mountedVolume := range rc.actualStateOfWorld.GetAllMountedVolumes() {
 		if !rc.desiredStateOfWorld.PodExistsInVolume(mountedVolume.PodName, mountedVolume.VolumeName) {
+			if !rc.shouldPodRuntimeBeRemoved(types.UID(mountedVolume.PodName)) {
+				klog.V(2).InfoS(mountedVolume.GenerateMsgDetailed("Starting operationExecutor.UnmountVolume", ""))
```
Comment (249043822) [inline pkg/kubelet/volumemanager/reconciler/reconciler.go:188]: This log looks confusing, I think this expects to not to umountVolume

File: pkg/volume/plugins.go [hunk 2]
```diff
@@ -160,6 +160,9 @@ type VolumePlugin interface {
 	// this to update the contents of the volume.
 	RequiresRemount(spec *Spec) bool
 
+	// Mountable returns true if this plugin requires mount calls
+	Mountable() bool
```
Comment (249043822) [inline pkg/volume/plugins.go:164]: does this interface relate with the issue which would be fixed by this PR?

**Comment by yingchunliu-zte** [comment]
@gnufied

**Comment by dims** [comment]
If you still need this PR then please rebase, if not, please close the PR

**Comment by jingxu97** [comment]
@jsafrane Without your recent changes in this logic, is this fix still needed?

**Comment by jsafrane** [comment]
Indeed, it should be better now.

@yingchunliu-zte, we have rewritten volume reconstruction in https://github.com/kubernetes/kubernetes/pull/113262, now it should not be possible that volumes are unmounted when kubelet can't talk to the API server. At least in theory. Please report a new issue / new PR when it happens.

/close

**Pull request closed without merging**